db/db_codes_test.go

```diff
@@ -0,0 +1,39 @@
+package db
+
+import (
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/jinzhu/gorm"
+	"github.com/stretchr/testify/assert"
+)
+
+var now = time.Now()
+
+var code = ConnectionCodes{
+	ID:               1,
+	ConnectionString: "2222222",
+	IsUsed:           false,
+	DateCreated:      &now,
+}
+
+func TestCodeGet(t *testing.T) {
+	db, mock, err := sqlmock.New()
+
+	if err != nil {
+		t.Fatalf("an error '%s' was not expected when opening a stub database connection", err)
+	}
+	defer db.Close()
+	gorm.Open("postgres", db)
+
+	rows := sqlmock.NewRows([]string{"connection_string", "date_created", "is_used", "date_created"}).AddRow(code.ID, code.ConnectionString, code.IsUsed, code.DateCreated)
+
+	mock.ExpectQuery(regexp.QuoteMeta(
+		`SELECT connection_string, date_created FROM connectioncodes WHERE is_used = ? ORDER BY id DESC LIMIT 1`)).
+		WithArgs(false).
+		WillReturnRows(rows)
```

Comment:
not exactly sure how this is working, are we able to call the endpoints/functions directly because it seems like we're directly adding to the db where we should be importing the connection_string functions and seeing if they function correctly.

I think if we break the connection string functionality this test will still pass, please correct me if I'm wrong since I'm not too familiar with testing in golang

Reply:
we can leave this for now closing